Networking/Networking/Model/ShippingLabel/ShippingLabelAddressValidationResponse.swift

```diff
@@ -2,26 +2,29 @@ import Foundation
 
 /// Represents Shipping Label Address that has been validated or that generated an error.
 ///
-public struct ShippingLabelAddressValidationResponse: Equatable, GeneratedFakeable {
-    public let address: ShippingLabelAddress?
-    public let errors: ShippingLabelAddressValidationError?
-
-    /// When sending an address to normalize to the server, if the response has the is_trivial_normalization property set to true,
-    /// then the normalized address will be automatically accepted without user intervention.
-    /// As its name indicates, that flag will be set when the changes made by the address normalizator were trivial,
-    /// such as adding the +4 portion to a ZIP code, or changing capitalization, or changing street to st for example.
-    ///
-    public let isTrivialNormalization: Bool?
+/// Only used internally for JSON decoding since the response might contain validation errors.
+/// For public consumption, we'll convert those to a `ShippingLabelAddressValidationError`, and expose
+/// a `ShippingLabelAddressValidationSuccess` instead if there were no errors.
+///
+internal struct ShippingLabelAddressValidationResponse: Equatable {
+    let result: Result<ShippingLabelAddressValidationSuccess, ShippingLabelAddressValidationError>
 
-    public init(address: ShippingLabelAddress?, errors: ShippingLabelAddressValidationError?, isTrivialNormalization: Bool?) {
-        self.address = address
-        self.errors = errors
-        self.isTrivialNormalization = isTrivialNormalization
+    init(address: ShippingLabelAddress?, errors: ShippingLabelAddressValidationError?, isTrivialNormalization: Bool?) {
+        if let errors = errors {
+            result = .failure(errors)
+        } else if let address = address,
+                  let isTrivialNormalization = isTrivialNormalization {
+            result = .success(.init(address: address, isTrivialNormalization: isTrivialNormalization))
+        } else {
+            // This case should never happen, but that's not guaranteed.
+            // We'll treat the absence of both an address and errors as an error with no message.
```

Comment:
In reality it can happen for all the errors related to HTTP, like all the errors `500`

Reply:
I don't think so, since this initializer is only ever called from the JSON decoding one, which means we did get a JSON response and not a network error.